src/tests/tracing/eventlistener/EventListenerThreadPool.cs

```diff
@@ -0,0 +1,80 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics.Tracing;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Tracing.Tests
+{
+    internal sealed class RuntimeEventListener : EventListener
+    {
+        public volatile int TPWorkerThreadStartCount = 0;
+        public volatile int TPWorkerThreadStopCount = 0;
+        public volatile int TPWorkerThreadWaitCount = 0;
+
+    public ManualResetEvent TPWaitEvent = new ManualResetEvent(false);
+        
+        protected override void OnEventSourceCreated(EventSource source)
+        {
+            if (source.Name.Equals("Microsoft-Windows-DotNETRuntime"))
+            {
+                EnableEvents(source, EventLevel.Informational, (EventKeywords)0x10000);
+            }
+        }
+
+        protected override void OnEventWritten(EventWrittenEventArgs eventData)
+        {
+            if (eventData.EventName.Equals("ThreadPoolWorkerThreadStart"))
+            {
+                Interlocked.Increment(ref TPWorkerThreadStartCount);
+                TPWaitEvent.Set();
+            }
+            else if (eventData.EventName.Equals("ThreadPoolWorkerThreadStop"))
+            {
+                Interlocked.Increment(ref TPWorkerThreadStopCount);
+                TPWaitEvent.Set();
+            }
+            else if (eventData.EventName.Equals("ThreadPoolWorkerThreadWait"))
+            {
+                Interlocked.Increment(ref TPWorkerThreadWaitCount);
+                TPWaitEvent.Set();
+            }
+        }
+    }
+
+    class EventListenerThreadPool
+    {
+        static int Main(string[] args)
+        {
+            using (RuntimeEventListener listener = new RuntimeEventListener())
+            {
+                int someNumber = 0;
+                Task[] tasks = new Task[100];
+                for (int i = 0; i < tasks.Length; i++) 
+                {
+                    tasks[i] = Task.Run(() => { someNumber += 1; });
+                }
+
+                listener.TPWaitEvent.WaitOne(TimeSpan.FromMinutes(3));
```

Comment:
Can we make this a more aggressive timeout? Perhaps 30s?

Reply:
We've run in to issues before where unnecessarily short timeouts mean the test fails in things like GCStress on arm machines. The test shouldn't ever hit the timeout, it's just there to not tank the whole test leg if it fails for some reason